src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -128,23 +127,30 @@ function Base.show(io::IO, ::MIME"text/plain", semi::SemidiscretizationHyperboli
     summary_line(io, "mesh", semi.mesh)
     summary_line(io, "equations", semi.equations |> typeof |> nameof)
     summary_line(io, "initial condition", semi.initial_condition)
-    summary_line(io, "boundary conditions", 2*ndims(semi))
-    if (semi.boundary_conditions isa Tuple ||
-        semi.boundary_conditions isa NamedTuple ||
-        semi.boundary_conditions isa AbstractArray)
-      bcs = semi.boundary_conditions
-    else
-      bcs = collect(semi.boundary_conditions for _ in 1:(2*ndims(semi)))
-    end
-    summary_line(increment_indent(io), "negative x", bcs[1])
-    summary_line(increment_indent(io), "positive x", bcs[2])
-    if ndims(semi) > 1
-      summary_line(increment_indent(io), "negative y", bcs[3])
-      summary_line(increment_indent(io), "positive y", bcs[4])
-    end
-    if ndims(semi) > 2
-      summary_line(increment_indent(io), "negative z", bcs[5])
-      summary_line(increment_indent(io), "positive z", bcs[6])
+    if semi.boundary_conditions isa Dict
+      summary_line(io, "boundary conditions", length(semi.boundary_conditions))
+      for (boundary_name, boundary_condition_type) in semi.boundary_conditions
+        summary_line(increment_indent(io), boundary_name, boundary_condition_type)
+      end
+    else # non dictionary boundary conditions container
+      summary_line(io, "boundary conditions", 2*ndims(semi))
+      if (semi.boundary_conditions isa Tuple ||
+          semi.boundary_conditions isa NamedTuple ||
+          semi.boundary_conditions isa AbstractArray)
+        bcs = semi.boundary_conditions
+      else
+        bcs = collect(semi.boundary_conditions for _ in 1:(2*ndims(semi)))
+      end
+      summary_line(increment_indent(io), "negative x", bcs[1])
+      summary_line(increment_indent(io), "positive x", bcs[2])
+      if ndims(semi) > 1
+        summary_line(increment_indent(io), "negative y", bcs[3])
+        summary_line(increment_indent(io), "positive y", bcs[4])
+      end
+      if ndims(semi) > 2
+        summary_line(increment_indent(io), "negative z", bcs[5])
+        summary_line(increment_indent(io), "positive z", bcs[6])
+      end
```

Comment:
This starts to become quite ugly but I can live with it for now

Reply:
I agree it is ugly. This is an artifact of wanting to output the boundary condition dictionary quickly. Once I actually get my external state functionality working we can iterate on the best way to show the dictionary information (assuming we even keep it as a dictionary)

Reply:
I added it to the list in https://github.com/trixi-framework/Trixi.jl/issues/542.